students/Matt_Hudgins/Lesson05/mailroom_3.py

```diff
@@ -0,0 +1,132 @@
+donors['Carl Grimes'] = [72.00, 10.00, 88.00]
+donors['Morgan Jones'] = [68.00, 10.00, 98.00]
+
+
+def donor_list():
+    '''This function prints a list of all current donors'''
+    for i in donors:
+        print(i)
+    return
+
+
+def adding_a_donor(newname):
+    '''This function adds a name to the donor list'''
+    donors[newname] = []
+    adding_a_donation(newname)
+    return
+
+
+def adding_a_donation(newname):
+    '''This function adds a donation to the the list'''
+    donation_amount = round(float(input(f'Enter a Donation amount for\n'
+    '{newname}\n')), 2)
+    if newname in donors.keys():
+        donors[newname].append(donation_amount)
+    elif newname not in donors:
+        donors[newname] = [donation_amount]
+    print(donors)
+    return
+
+
+def send_letter(newname, amount):
+        letter = 'Dear {},\n Thank you for your donation of ${}'
+        print(letter.format(newname, amount))
+        print('\n')
+        return
+
+
+def create_report():
+    '''This function creates a report'''
+    headers = ['Donor Name', 'Total Given', 'Num Gifts', 'Average Gift']
+    print('{:20} | {:20} | {:20} | {:20}'.format(*headers))
+    for name, donations in donors.items():
+        total = sum(donations)
+        num = len(donations)
+        avg = total / num
+        spacing = '{:20} | {:20} | {:20} | {:20}'
+        print(spacing.format(name, total, num, avg))
+    print('\n')
+    return
+
+def letter_to_all():
+        big_letter = ('Dear {},\n\n Thank you for you donation of {}\n'
+        'your money will go a long way\n')
+        for donor, donations in donors.items():
+            print(big_letter.format(donor, sum(donations)))
+            filename = donor.replace(' ', ' ') + '.txt'
+            save = open(filename, 'w')
+            save.write(big_letter.format(donor, sum(donations)))
+            save.close()
+        return
+
+
+def main_menu():
+    """main block"""
+    while True:
+        prompt = 'Please enter a donor\'s name or type list:\n'
+        entry = input(prompt)
+        if entry.lower() == "list":
+            donor_list()
+        elif entry.lower() == "quit":
+            break
+        else:
+            try:
+                assert entry in donors
+                adding_a_donation(entry)
+            except AssertionError:
+                print('{entry} is not in list')
+                adding_a_donor(entry)
+
+
+main_prompt = ('\nDonation Update\n'
+               'Please enter a selection\n'
+               '1. Send a Thank You\n'
+               '2. Create a report\n'
+               '3. Mail all donors\n'
+               '0. Exit menu\n'
+               )
+
+main_dispatch = {'1': main_menu,
+                 '2': create_report,
+                 '3': letter_to_all,
+                 '0': quit
+                 }
+
+
+def menu(prompt, dispatch):
+    while True:
+        entry = input(prompt).lower()
+        try:
+            if dispatch.get(entry)() == "exit menu":
```

Comment:
This `if` condition is never actually checking anything. Your `main_dispatch` functions are called in the `if` statement before the `==` operator, so it's never checking the return value against 'exit menu'. And when a user enters '0', it's calling Python built-in `quit` function, rather than your `quit` function defined below.

I would rename your `quit` function to something that doesn't clash with built in, and move it above the `main_dispatch` dict. Then you can use dict key bracket access in your `try` block to call the functions (e.g. `dispatch['1']()`) and catch a `KeyError` with your `except`clause if the user enters any value that isn't a key in your dict